providers/base/tests/test_stress_ng_test.py

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Ltd.
+# Written by:
+#   Pedro Avalos Jimenez <pedro.avalosjimenez@canonical.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+from subprocess import CalledProcessError, TimeoutExpired
+from unittest.mock import MagicMock, mock_open, patch
+
+from stress_ng_test import main, num_numa_nodes, swap_space_ok
+
+
+class TestMemoryFunctions(unittest.TestCase):
+    @patch("stress_ng_test.run")
+    def test_num_numa_nodes_success(self, run_mock):
+        run_mock.return_value = MagicMock()
+        run_mock.return_value.stdout = b"available: 2 nodes (0-1)"
+        self.assertEqual(num_numa_nodes(), 2)
+
+    @patch("stress_ng_test.run", side_effect=OSError)
+    def test_num_numa_nodes_failure(self, run_mock):
+        self.assertEqual(num_numa_nodes(), 1)
+
+    @patch(
+        "stress_ng_test.psutil.swap_memory", return_value=MagicMock(total=1)
+    )
+    def test_swap_space_ok_success(self, psutil_swap_memory_mock):
+        self.assertTrue(swap_space_ok(0))
+
+    @patch("stress_ng_test.run")
+    @patch("stress_ng_test.os.chmod")
+    @patch("stress_ng_test.open", new_callable=mock_open)
+    @patch(
+        "stress_ng_test.psutil.swap_memory",
+        side_effect=[MagicMock(total=0), MagicMock(total=1073741824)],
+    )
+    def test_swap_space_ok_create_swap(
```

Comment:
This takes a while because it still goes through the loop to write to the fake file (also, quite a bit of ram as the mock records the writes). Add:
```
    @patch("stress_ng_test.range", return_value=[0])
```
to avoid that

Reply:
Resolved in fba62fd